addons/binding/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/handler/PHCBridgeHandler.java

```diff
@@ -0,0 +1,710 @@
+                }
+
+                byte size = 0;
+                result.add(buffer[pos]);
+                size = (byte) (buffer[pos] & 0x7F);
+
+                if (size < 4) {
+                    byte[] inByte = new byte[size + 4];
+                    inByte[0] = moduleAddress;
+                    inByte[1] = buffer[pos++];
+
+                    for (int i = 2; i < size + 4; i++) {
+                        inByte[i] = buffer[pos++];
+                        result.add(inByte[i]);
+                    }
+
+                    if (!repeat.contains(Arrays.toString(inByte))) { // avoids multiple processing of the same message
+                        processIncomingMessage(inByte, size);
+                        repeat.add(Arrays.toString(inByte));
+                    }
+                }
+
+            } catch (ArrayIndexOutOfBoundsException e) { // if the message is interrupted at the end of the buffer
+                if (messageStart < 0) {
+                    messageFragment = new byte[buffer.length - pos + 1];
+                    System.arraycopy(buffer, messageStart, messageFragment, 0, messageFragment.length);
+                }
+            }
+        }
+        return result.toArray(new Byte[result.size()]);
+    }
+
+    private void processQueueLoop() {
+        while (true) {
+            try {
+                long limit = System.currentTimeMillis();
+
+                ConcurrentNavigableMap<Long, QueueObject> subQueue = queue.subMap(0L, true, limit, true);
+
+                for (Long key : subQueue.keySet()) {
+                    QueueObject qo = subQueue.get(key);
+                    queue.remove(key);
+                    if (qo.getCounter() < SEND_RETRY_COUNT && !qo.getCommand().equals(
+                            isChannelOutputStateSet(qo.getModuleType(), qo.getmoduleAddress(), qo.getChannel()))) {
+                        qo.increaseCounter();
+
+                        switch (qo.getModuleType()) {
+                            case PHCBindingConstants.CHANNELS_AM:
+                                sendAm(qo.getmoduleAddress(), qo.getChannel(), qo.getCommand());
+                                break;
+                            case PHCBindingConstants.CHANNELS_EM_LED:
+                                sendEm(qo.getmoduleAddress(), qo.getChannel(), qo.getCommand());
+                                break;
+                            case PHCBindingConstants.CHANNELS_JRM:
+                                sendJRM(qo.getmoduleAddress(), qo.getChannel(), qo.getCommand());
+                                break;
+                        }
+
+                        queue.put(System.currentTimeMillis() + SEND_RETRY_TIME, qo);
+                    } else if (qo.getCounter() >= SEND_RETRY_COUNT
+                            && !qo.getModuleType().equals(PHCBindingConstants.CHANNELS_JRM)) {
+                        // Can´t process the acknowledgement of JRM yet.
+                        // TODO: Thing state to offline?
```

Comment:
I think the serial communication isn't enough reliable. I wrote more detail in the TODO.